one-ui/src/test/java/com/marklogic/hub/oneui/controllers/SavedQueriesControllerTest.java

```diff
@@ -0,0 +1,215 @@
+        Assertions.assertEquals("test-data-hub-developer", responseBody.get("savedQuery").get("owner").asText());
+        Assertions.assertEquals("test-data-hub-developer", responseBody.get("savedQuery").get("systemMetadata").get("createdBy").asText());
+    }
+
+    @Test
+    void testModifyExistingQuery() {
+        JsonNode firstResponse = savedQueriesController.saveQueryDocument(queryDoc).getBody();
+        ObjectNode savedQueryNode = (ObjectNode) firstResponse.get("savedQuery");
+        String id = savedQueryNode.get("id").asText();
+        savedQueryNode.put("name", "modified-name");
+        ResponseEntity<JsonNode> modifiedResponse = savedQueriesController.updateQueryDocument(firstResponse);
+        JsonNode modifiedResponseBody = modifiedResponse.getBody();
+
+        Assertions.assertEquals(200, modifiedResponse.getStatusCodeValue());
+        Assertions.assertNotNull(modifiedResponseBody);
+        Assertions.assertEquals(id, modifiedResponseBody.get("savedQuery").get("id").asText());
+        Assertions.assertEquals("test-data-hub-developer", modifiedResponseBody.get("savedQuery").get("owner").asText());
+        Assertions.assertEquals("modified-name", modifiedResponseBody.get("savedQuery").get("name").asText());
+    }
+
+    @Test
+    void testCollectionsAndPermissionOnSavedQuery() {
+        JsonNode response = savedQueriesController.saveQueryDocument(queryDoc).getBody();
+        String docUri = "/saved-queries/" + response.get("savedQuery").get("id").asText() + ".json";
+
+        DocumentMetadataHandle metadataHandle = new DocumentMetadataHandle();
+        GenericDocumentManager docMgr = getFinalGenericDocumentManager(DatabaseKind.FINAL);
+        docMgr.readMetadata(docUri, metadataHandle);
+        DocumentMetadataHandle.DocumentPermissions permissions = metadataHandle.getPermissions();
+        DocumentMetadataHandle.DocumentCollections collections = metadataHandle.getCollections();
+        Set<DocumentMetadataHandle.Capability> readQueryCap = new HashSet<>(Arrays.asList(DocumentMetadataHandle.Capability.READ));
+        Set<DocumentMetadataHandle.Capability> writeQueryCap = new HashSet<>(Arrays.asList(DocumentMetadataHandle.Capability.UPDATE));
+        Set<String> expectedCollections = new HashSet<>(Arrays.asList("http://marklogic.com/data-hub/saved-query"));
+
+        Assertions.assertNotNull(permissions.get("data-hub-saved-query-writer"));
+        Assertions.assertNotNull(permissions.get("data-hub-saved-query-reader"));
+        Assertions.assertTrue(readQueryCap.equals(permissions.get("data-hub-saved-query-reader")));
+        Assertions.assertTrue(writeQueryCap.equals(permissions.get("data-hub-saved-query-writer")));
+        Assertions.assertTrue(expectedCollections.equals(collections));
+    }
+
+    @Test
+    void testCollectionsAndPermissionOnModifiedQuery() {
+        JsonNode firstResponse = savedQueriesController.saveQueryDocument(queryDoc).getBody();
```

Comment:
Same thing, just add these assertions to the test for modifying a query. 

Really, you can cover both things - creating a query and modifying it - in the same test.